src/components/CharactersPage/CharactersPage.js

```diff
@@ -24,54 +20,29 @@ const useStyles = makeStyles(() => ({
 export default function CharactersPage() {
   const classes = useStyles()
 
-  const router = useRouter()
-  const { params } = router.query
-
-  const isFormOpen = useMemo(() => {
-    return params && Object.values(formModeEnum).includes(params[0])
-  }, [params])
-
-  const handleOpenForm = (id) => {
-    const params = id ? [formModeEnum.edit, id] : [formModeEnum.new]
-
-    router.push({
-      pathname: "/characters/[[...params]]",
-      query: { params },
-    })
-  }
-
-  const handleCloseForm = () => {
-    router.push("/characters/[[...params]]")
-  }
-
-  useEffect(() => {
-    if (!params) return
-
-    if (params.length > 2) {
-      handleCloseForm()
+  const { error, loading, data, fetchMore, refetch } = useQuery(
+    GET_ALL_CHARACTERS,
+    {
+      fetchPolicy: "network-only",
+      variables: {
+        first: 10,
+      },
     }
+  )
 
-    if (params.length > 0 && !isFormOpen) {
-      handleCloseForm()
-    }
-  }, [params, isFormOpen])
+  if (error) return <div>{error.message}</div>
+  if (loading) return <div>loading...</div>
 
   return (
     <div className={classes.body}>
-      <RightSideDrawer
-        isOpen={isFormOpen}
-        onClose={handleCloseForm}
-      >
-        <CharacterForm onClose={handleCloseForm} />
-      </RightSideDrawer>
       <Typography variant="h2">Star Wars</Typography>
-      <div className={classes.button}>
-        <AddNewButton
-          label="Add New Character"
-          onClick={() => handleOpenForm()}
-        />
-      </div>
-      <CharactersTable onRowClick={({ id }) => handleOpenForm(id)} />
+
+      <CharactersTable
+        onRowClick={() => console.log("Table clicked")}
+        data={data}
+        fetchMore={fetchMore}
+        refetch={refetch}
```

Comment:
Please group your props together based on their type, so functions should be grouped together at the bottom and should be prefixed with `on`, for example `onFetchMore`